modules/infra/datapath/control_input.c

```diff
@@ -74,6 +77,21 @@ control_input_process(struct rte_graph *graph, struct rte_node *node, void **, u
 	return n;
 }
 
+static int control_input_init(const struct rte_graph *graph, struct rte_node *node) {
+	node->ctx_ptr = gr_pktmbuf_pool_get(graph->socket, RTE_GRAPH_BURST_SIZE);
+
+	if (node->ctx_ptr == NULL) {
+		return errno_log(-errno, "gr_pktmbuf_pool_get(control_input)");
+	}
```

Comment:
This should be just `errno` without a negative sign. `errno_log` expects a positive `errno` value.

Also please remove the braces here.

```c
static int control_input_init(const struct rte_graph *graph, struct rte_node *node) {
        node->ctx_ptr = gr_pktmbuf_pool_get(graph->socket, RTE_GRAPH_BURST_SIZE);
        if (node->ctx_ptr == NULL)
                return errno_log(-errno, "gr_pktmbuf_pool_get(control_input)");
        return 0;
}
```

Last nag, I promise :)

Reply:
Should be fixed now :)